src/frontend/xgboost/xgboost_json.h

```diff
@@ -0,0 +1,376 @@
+
+  bool StartObject() {
+    return StartObject(
+        std::integral_constant<bool, std::is_base_of<OutputHandler<ElemType>,
+                                                     HandlerType>::value>{});
+  }
+};
+
+/*! \brief handler for TreeParam objects from XGBoost schema*/
+class TreeParamHandler : public OutputHandler<int> {
+ public:
+  using OutputHandler<int>::OutputHandler;
+
+  bool String(const char *str, std::size_t length, bool copy);
+};
+
+/*! \brief handler for RegTree objects from XGBoost schema*/
+class RegTreeHandler : public OutputHandler<treelite::Tree> {
+ public:
+  using OutputHandler<treelite::Tree>::OutputHandler;
+  bool StartArray();
+
+  bool StartObject();
+
+  bool Uint(unsigned u);
+
+  bool EndObject(std::size_t memberCount);
+
+ private:
+  std::vector<double> loss_changes;
+  std::vector<double> sum_hessian;
+  std::vector<double> base_weights;
+  std::vector<int> leaf_child_counts;
+  std::vector<int> left_children;
+  std::vector<int> right_children;
+  std::vector<int> parents;
+  std::vector<int> split_indices;
+  std::vector<double> split_conditions;
+  std::vector<bool> default_left;
+  int num_nodes;
+};
+
+/*! \brief handler for GBTreeModel objects from XGBoost schema*/
+class GBTreeModelHandler : public OutputHandler<treelite::Model> {
+  using OutputHandler<treelite::Model>::OutputHandler;
+  bool StartArray();
+  bool StartObject();
+};
+
+/*! \brief handler for GradientBoosterHandler objects from XGBoost schema*/
+class GradientBoosterHandler : public OutputHandler<treelite::Model> {
+ public:
+  using OutputHandler<treelite::Model>::OutputHandler;
+  bool Uint(unsigned u);
+  bool String(const char *str, std::size_t length, bool copy);
+  bool StartObject();
+};
+
+/*! \brief handler for ObjectiveHandler objects from XGBoost schema*/
+class ObjectiveHandler : public OutputHandler<std::string> {
+  using OutputHandler<std::string>::OutputHandler;
+
+  bool StartObject();
+
+  bool String(const char *str, std::size_t length, bool copy);
+};
+
+/*! \brief handler for LearnerParam objects from XGBoost schema*/
+class LearnerParamHandler : public OutputHandler<treelite::Model> {
+ public:
+  using OutputHandler<treelite::Model>::OutputHandler;
+  bool String(const char *str, std::size_t length, bool copy);
+};
+
+/*! \brief handler for Learner objects from XGBoost schema*/
+class LearnerHandler : public OutputHandler<treelite::Model> {
+ public:
+  using OutputHandler<treelite::Model>::OutputHandler;
+  bool StartArray();
+  bool StartObject();
+  bool EndObject(std::size_t memberCount);
+
+ private:
+  std::string objective;
+};
+
+/*! \brief handler for XGBoostModel objects from XGBoost schema*/
+class XGBoostModelHandler : public OutputHandler<treelite::Model> {
+ public:
+  using OutputHandler<treelite::Model>::OutputHandler;
+  bool StartArray();
+  bool StartObject();
+  bool EndObject(std::size_t memberCount);
+
+ private:
+  std::vector<unsigned> version;
```

Comment:
```suggestion
  std::vector<unsigned> m_version;
```
to be consistent with `m_output`

Reply:
I used the `m_` prefix on `m_output` just to distinguish it from the `output` argument. Unless we're using `m_` consistently throughout the rest of the codebase, I'd rather remove it from `m_output` and use something else, since I'm generally not a fan of polluting the variable name with an indication that it's a member variable if that's clear from the rest of the code. I'll do a quick check for how we're doing this elsewhere, and either match the rest of the codebase or rename `m_output`.

Reply:
@wphicks Got it. Then let's rename `m_output`.